dist/gcc-compatible/internal/Hacl_Hash_Base.h

```diff
@@ -22,8 +22,8 @@
  */
 
 
-#ifndef __internal_Hacl_Hash_Blake2_H
-#define __internal_Hacl_Hash_Blake2_H
+#ifndef __internal_Hacl_Hash_Base_H
```

Comment:
Is this really generic for all hashing? Looks like it's only used in blake2.
And if it's used across multiple algorithms, I wonder if it's not breaking the ability to use only a subset of algorithms.

Reply:
@msprotz what should we do?
The line that defines this bundle is  `-bundle Hacl.Hash.Definitions=Hacl.Hash.*[rename=Hacl_Hash_Base]`, how is it that only blake2 uses it?
It also seems like this PR is adding a lot of code in `Hacl_Hash_Base.c`.